src/http/content.cr

```diff
@@ -60,7 +60,7 @@ module HTTP
 
     def read(slice : Bytes) : Int32
       ensure_send_continue
-      @io.read(slice)
+      @io.read(slice).to_i32
```

Comment:
Are there `@io.read` that does not return `Int32`? It seems odd to add all the `to_i32` on methods that should already be returning `Int32` at a first glance.

Reply:
Yes: if a shard inherits IO and returns another integer type.

Reply:
(this was the main issue that happened, and that's why the original type restrictions had to be reverted... that's why I'm still not convinced that forcing return types here isn't going to break something else, but we'll see)

Reply:
Well, it could technically break code that was explicitly expecting an `Int64` return type, for example. But I don't think that's likely. Almost all source implementations in stdlib return `Int32`. So it would need to be very specific and selective code that doesn't at least have `Int32` return type as a union member.

Reply:
Oh, ok, the abstract `IO#read` does not have a return type restriction and so implementors can choose any return type. Eventually on 2.0 we can make that return type explicit and the to_i32 can go away since the compiler will ensure that all implementations honor the return type.

👍 then.